erts/emulator/beam/global.h

```diff
@@ -642,6 +642,14 @@ do {						\
 
 #define WSTACK_IS_STATIC(s) (s.wstart == WSTK_DEF_STACK(s))
 
+#define WSTACK_RESERVE(s, push_cnt)             \
+do {						\
+    UWord *wend = s.wend - (push_cnt);          \
+    if (s.wsp > wend || wend > s.wend) {        \
+	erl_grow_wstack(&s, (push_cnt));        \
```

Comment:
Is it not even better with
`if (s.wend - s.wsp < (SWord)push_cnt)
`
Pointer arithmetic outside allocated areas is undefined.

Also, same bug in ESTACK_RESERVE.
We should probably merge the two some time. WSTACK is a residue from halfword emulator.

Reply:
I agree with @sverker, but sholdn't it be:

```
if (s.wend - s.wsp < (UWord)push_cnt)
```

because the difference between two pointers is unsigned?

Reply:
Pointer subtraction is signed.
If wsp is larger than wend (which I hope it never can be) then a signed comparison could save the day and trigger a stack growth.